website_medical_patient/views/patients_template.xml

```diff
@@ -72,7 +54,114 @@
+				<input type="hidden" name="parent_id" t-att-value="patient.parent_id.id or partner.id" />
+				<input type="hidden" name="id" t-att-value="patient.id or 0" />
+
+				<div class="row o_website_portal_details">
+					<div class="col-xs-12 patient-details-form-content">
+						<div class="row patient-details-error-message">
+							<span id="o_website_form_result" />
+							<div t-if="error_message" class="alert alert-danger">
+								<t t-foreach="error_message" t-as="err">
+									<t t-esc="err"/><br />
+								</t>
+							</div>
+						</div>
+						<div class="row">
+							<div t-attf-class="form-field #{error.get('name') and 'has-error' or ''} o_website_form_required_custom">
+								<label class="control-label" for="name">
+									Full Name
+								</label>
+								<input type="text"
+									   name="name"
+									   class="form-control o_website_form_input"
+									   required=""
+									   t-att-value="request.params.get('name') or patient.name or ''" />
+							</div>
+						</div>
+						<div class="row">
+							<div t-attf-class="form-field #{error.get('phone') and 'has-error' or ''}">
+								<label class="control-label" for="phone">
+									Phone
+								</label>
+								<input type="tel"
+									   name="phone"
+									   class="form-control o_website_form_input"
+									   t-att-value="request.params.get('phone') or patient.phone or partner.phone or ''" />
+							</div>
+						</div>
+						<div class="row">
+							<div t-attf-class="o_datepicker form-field #{error.get('dob') and 'has-error' or ''} o_website_form_required_custom">
+								<label class="control-label label-optional" for="dob">
+									Birth Date
+								</label>
+								<input type="text"
+									   name="dob"
+									   required=""
+									   class="form-control o_website_form_date"
+									   t-att-value="request.params.get('dob') or _format_date(patient.dob) or ''" />
+							</div>
+						</div>
+						<div class="row">
+							<div t-attf-class="form-field #{error.get('gender') and 'has-error' or ''} o_website_form_required_custom">
+								<label class="control-label label-optional" for="gender">
+									Gender
+								</label>
+								<select name="gender" class="form-control o_website_form_input" required="">
+									<option value="">Gender...</option>
+									<t t-foreach="[('f', 'Female'), ('m', 'Male'), ]"
+									   t-as="gender">
+										<option t-att-value="gender[0]"
+												t-att-selected="gender[0] == (request.params.get('gender') or patient.gender)">
+											<t t-esc="gender[1]" />
+										</option>
+									</t>
+								</select>
+							</div>
+						</div>
+						<div class="row">
+							<div t-attf-class="form-field #{error.get('image') and 'has-error' or ''}">
+								<label class="control-label label-optional" for="image">
+									Profile Picture
+								</label>
+								<input type="binary"
```

Comment:
binary input isn't a thing. You could try the image widget, but I don't think that will work in front end. 

You form needs to have `enctype="multipart/form-data"` & your input `type="file"`